src/kube-runtime/test/test_image_checker.py

```diff
@@ -52,69 +61,145 @@ def wrapper(self, *args, **kwargs):
+                  status=http.HTTPStatus.OK)
+    responses.add(
+        responses.HEAD,
+        "https://openpai.azurecr.io/v2/{repo}/manifests/{tag}".format(
+            **image_info),
+        status=http.HTTPStatus.OK)
+
+
+def official_registry_request_callback(request, repo, options):
+    headers = request.headers
+    if "Authorization" in headers and headers["Authorization"].startswith(
+            "Bearer"):
+        if options and "image_not_found" in options and options[
+                "image_not_found"]:
+            return (http.HTTPStatus.NOT_FOUND, {}, None)
+        return (http.HTTPStatus.OK, {}, None)
+    scope = "repository:{}:pull".format(repo)
+    headers = {
+        "Www-Authenticate":
+        ("Bearer realm=\"https://auth.docker.io/token\","
+         "service=\"registry.docker.io\",scope=\"{}\"".format(scope))
+    }
+    return (http.HTTPStatus.UNAUTHORIZED, headers, None)
+
+
 class TestImageChecker(unittest.TestCase):
     def setUp(self):
         try:
             os.chdir(PACKAGE_DIRECTORY_COM)
         except Exception:  #pylint: disable=broad-except
             pass
-        self.config = {}
+        self.job_config = {}
+        self.secret = {}
 
-    @prepare_image_check("docker_image_no_user_tag.yaml")
-    def test_image_without_username_tag(self):
-        res = image_checker._is_docker_image_valid(self.config)
-        self.assertEqual(res, True)
-
-    @prepare_image_check("docker_image_no_tag.yaml")
-    def test_image_without_tag(self):
-        res = image_checker._is_docker_image_valid(self.config)
-        self.assertEqual(res, True)
+    @prepare_image_check("docker_official_image.yaml")
+    @responses.activate
+    def test_official_image(self):
+        image_checker = ImageChecker(self.job_config, self.secret)
+        image_info = image_checker._get_normalized_image_info()
+        add_official_registry_v2_response(image_info)
+        self.assertTrue(image_checker.is_docker_image_accessible())
 
     @prepare_image_check("docker_image_no_exist.yaml")
+    @responses.activate
     def test_image_not_exist(self):
-        res = image_checker._is_docker_image_valid(self.config)
-        self.assertEqual(res, False)
-
-    @prepare_image_check("docker_image_local_registry.yaml")
-    def test_image_local(self):
-        res = image_checker._is_docker_image_valid(self.config)
-        self.assertEqual(res, True)
+        image_checker = ImageChecker(self.job_config, self.secret)
+        image_info = image_checker._get_normalized_image_info()
+        add_official_registry_v2_response(image_info,
+                                          options={"image_not_found": True})
+        self.assertFalse(image_checker.is_docker_image_accessible())
+
+    @prepare_image_check("docker_image_acr_registry.yaml")
+    @responses.activate
+    def test_acr_image(self):
+        image_checker = ImageChecker(self.job_config, self.secret)
+        image_info = image_checker._get_normalized_image_info()
```

Comment:
why not add the common lines to decorator

Reply:
fixed

---
In reply to: [368789410](https://github.com/microsoft/pai/pull/4156#discussion_r368789410) [](ancestors = 368789410)